core/src/avm2/globals/string.rs

```diff
@@ -408,14 +408,40 @@ fn split<'gc>(
                     .into(),
             );
         }
-        if delimiter
+
+        let this = Value::from(this).coerce_to_string(activation)?;
+
+        if let Some(mut regexp) = delimiter
             .as_object()
-            .map(|o| o.as_regexp().is_some())
-            .unwrap_or(false)
+            .as_ref()
+            .and_then(|o| o.as_regexp_mut(activation.context.gc_context))
         {
-            log::warn!("string.split(regex) - not implemented");
+            let orig_flags = regexp.flags();
+            let orig_index = regexp.last_index();
+            regexp.set_flags(orig_flags | RegExpFlags::GLOBAL);
```

Comment:
Are you sure the flags need to be changed? I don't see any evidence for this in [avmplus](https://github.com/adobe/avmplus).

Reply:
So this is a design question. We either leave regex alone (and have to temporarily modify it) or implement a special method inside the regex object for splitting.

I changed the PR to use the latter, also found a bug in Flash and changed the implementation to match it.